src/DynamoCore/Core/NodeGraph.cs

```diff
@@ -114,14 +115,39 @@ private static IEnumerable<NoteModel> LoadNotesFromXml(XmlDocument xmlDoc)
         /// </summary>
         /// <param name="xmlDoc">An XmlDocument representing a serialized Dynamo workspace.</param>
         /// <param name="nodeFactory">A NodeFactory, used to load and instantiate nodes.</param>
+        /// <param name="elementResolver"></param>
         /// <returns></returns>
-        public static NodeGraph LoadGraphFromXml(XmlDocument xmlDoc, NodeFactory nodeFactory)
+        public static NodeGraph LoadGraphFromXml(XmlDocument xmlDoc, NodeFactory nodeFactory, out ElementResolver elementResolver)
         {
+            elementResolver = LoadElementResolver(xmlDoc);
+
             var nodes = LoadNodesFromXml(xmlDoc, nodeFactory).ToList();
             var connectors = LoadConnectorsFromXml(xmlDoc, nodes.ToDictionary(node => node.GUID)).ToList();
             var notes = LoadNotesFromXml(xmlDoc).ToList();
 
             return new NodeGraph { Nodes = nodes, Connectors = connectors, Notes = notes };
         }
+
+        private static ElementResolver LoadElementResolver(XmlDocument xmlDoc)
+        {
+            var nodes = xmlDoc.GetElementsByTagName("NamespaceResolutionMap");
+            
+            if (nodes.Count == 0)
+            {
+                // no namespace resolution map information exists in the DYN file
+                return null;
```

Comment:
I recommend an `ElementResolver` so that none of the downstream operations need to check it against `null`.